provider/datagen/src/transform/cldr/units/helpers.rs

```diff
@@ -0,0 +1,227 @@
+    }
+
+    let mut split = number.split('.');
+    if split.clone().count() > 2 {
+        return false;
+    }
+    let whole = split.next().unwrap_or("0");
+    let fractional = split.next().unwrap_or("0");
+    if whole.chars().any(|c| !c.is_ascii_digit()) {
+        return false;
+    }
+    if fractional.chars().any(|c| !c.is_ascii_digit()) {
+        return false;
+    }
+    true
+}
+
+/// Converts a rational number represented as a string into a tuple of (numerator, denominator).
+pub fn to_fractional(number: &str) -> Result<(u64, u64), DataError> {
+    if !check_if_decimal_number(number) {
+        return Err(DataError::custom("the number is not a decimal number"));
+    }
+
+    let mut split = number.split('.');
+    let whole = split.next().unwrap_or("0");
+    let fractional = split.next().unwrap_or("0");
+    let mut denominator = 1;
+    for _ in 0..fractional.len() {
+        denominator *= 10;
+    }
+    let numerator = whole.to_string() + fractional;
+    let numerator = numerator.parse::<u64>().unwrap_or(0);
+    let gcd = gcd(numerator, denominator);
+    Ok((numerator / gcd, denominator / gcd))
+}
+
+/// Checks if a string represents a scientific notation number.
+pub fn check_if_scientific_notation_number(number: &str) -> bool {
+    let mut split = number.split('E');
+    if split.clone().count() > 2 {
+        return false;
+    }
+
+    let base = split.next().unwrap_or("0");
+    let exponent = split.next().unwrap_or("0");
+    check_if_decimal_number(base) && check_if_decimal_number(exponent)
+}
+
+/// Converts a scientific notation number represented as a string into a tuple of (numerator, denominator).
+pub fn convert_scientific_notation_number_to_fractional(number: &str) -> Result<(u64, u64), DataError> {
+    if !check_if_scientific_notation_number(number) {
+        return Err(DataError::custom(
+            "the number is not a scientific notation number",
+        ));
+    }
+
+    let mut split = number.split('E');
+    let base = split.next().unwrap_or("0");
+    let exponent = split.next().unwrap_or("0");
+
+    let (mut numerator, mut denominator) = to_fractional(base).unwrap();
+    let exponent = exponent.parse::<i32>().unwrap();
+    if exponent > 0 {
+        for _ in 0..exponent {
+            numerator *= 10;
+        }
+    } else {
+        for _ in 0..(-exponent) {
+            denominator *= 10;
+        }
+    }
+    let gcd = gcd(numerator, denominator);
+    Ok((numerator / gcd, denominator / gcd))
+}
+
+#[test]
+fn test_remove_whitespace() {
+    let input = "He  llo Wo rld!";
+    let expected = "HelloWorld!";
+    let actual = remove_whitespace(input);
+    assert_eq!(expected, actual);
+}
+
+#[test]
+fn test_split_string() {
+    let input = "Hello,World!/in/ICU4X";
+    let expected = vec![
+        "Hello".to_string(),
+        "World!".to_string(),
+        "in".to_string(),
+        "ICU4X".to_string(),
+    ];
+    let actual = split_string(input, vec![",", "/"]);
```

Comment:
If your splitter are single characters only, the standard library has you covered:

```rust
input.split([',', '/'])
```

Allocation free